### PR TITLE
Tank vendor

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Adds tank vendor/elevator to all shipmaps except PoS
## Changelog
:cl:
add: Added tank vendor
/:cl:

### CODE REVIEW

**Comment by Dolth** [comment]
Why not pos?

**Comment by TiviPlus** [comment]
Pos has an open pr reworking it

**Review by Lumipharon — APPROVED** [review]

**Comment by novaepee** [comment]
https://github.com/tgstation/TerraGov-Marine-Corps/pull/15539

pos hotfix for tc is ready

File: code/modules/reqs/supply.dm [hunk 1]
```diff
@@ -618,3 +621,298 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
...
+		gun = new gun
+		GLOB.armored_gunammo[gun.type] = gun.accepted_ammo
+		qdel(gun)
+
+/datum/supply_ui/vehicles
+	tgui_name = "VehicleSupply"
+	shuttle_id = SHUTTLE_VEHICLE_SUPPLY
+	home_id = "vehicle_home"
+	/// current selected vehicles typepath
+	var/current_veh_type
```
Comment (BraveMole) [inline code/modules/reqs/supply.dm:696]: imagine being so lazy that you cannot type vehicle

File: code/modules/reqs/supply.dm [hunk 2]
```diff
@@ -618,3 +621,298 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
...
+					"name" = initial(mod.name),
+					"desc" = initial(mod.desc),
+					"type" = mod,
+					"isselected" = (current_gunner_mod == mod),
+				))
+	return data
+
+/datum/supply_ui/vehicles/ui_data(mob/living/user)
+	var/list/data = list()
+	if(supply_shuttle)
```
Comment (BraveMole) [inline code/modules/reqs/supply.dm:781]: if(!supply_shuttle)
   data["elevator"] = "MISSING!"
   return data

**Review by BraveMole — APPROVED** [review]

**Pull request merged**